azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/WarArtifactHandlerImpl.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.maven.webapp.handlers;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.apache.maven.model.Resource;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.codehaus.plexus.util.StringUtils;
+
+import com.google.common.io.Files;
+import com.microsoft.azure.management.appservice.WebApp;
+import com.microsoft.azure.maven.webapp.AbstractWebAppMojo;
+
+public class WarArtifactHandlerImpl implements ArtifactHandler  {
+
+    public static final String FILE_IS_NOT_WAR = "The deployment file is not a war typed file.";
+    private AbstractWebAppMojo mojo;
+
+    public static final String FIND_WAR_FILE_FAIL = "Failed to find the war file: '%s'";
+    public static final int DEFAULT_MAX_RETRY_TIMES = 3;
+    public static final String UPLOAD_FAILURE = "Failed to deploy the war file to server, " +
+            "retrying immediately (%d/%d)";
+
+    public WarArtifactHandlerImpl(final AbstractWebAppMojo mojo) {
+        this.mojo = mojo;
+    }
+
+    @Override
+    public void publish(final List<Resource> resources) throws Exception {
```

Comment:
resources not used for War?

Reply:
Yes. How about make resources as a member in which class needs it?

Reply:
Updated, get the resources from `mojo` and `mojo` is already a member of the Hander